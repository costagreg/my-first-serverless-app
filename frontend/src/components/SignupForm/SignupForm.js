import React from 'react'
import PropTypes from 'prop-types'
import useInput from '../../hooks/useInput'

import './SignupForm.scss'

const SignupForm = ({ signup, error }) => {
  const { emailValue, emailBindInput } = useInput('email')
  const { passwordValue, passwordBindInput } = useInput('password')
  const { confirmPasswordValue, confirmPasswordBindInput } = useInput(
    'password'
  )

  const handleSubmit = (evt) => {
    if (passwordValue === confirmPasswordValue) {
      signup(emailValue, passwordValue)
    } else {
      signupError("Password doesn't match")
    }
    evt.preventDefault()
  }

  return (
    <div className="signupForm">
        <form className="signupForm__formContainer" onSubmit={handleSubmit}>
          <input
            className="signupForm__email"
            placeholder="Email"
            type="email"
            {...emailBindInput}
            required
          />
          <input
            type="password"
            className="signupForm__password"
            placeholder="Password"
            {...passwordBindInput}
            required
          />
          <input
            type="password"
            className="signupForm__password"
            placeholder="Confirm Password"
            {...confirmPasswordBindInput}
            required
          />
          <div className="signupForm__buttons">
            <button className="signupForm__submit">Sign up</button>
          </div>
          {error}
        </form>
    </div>
  )
}

SignupForm.propTypes = {}

export default SignupForm
