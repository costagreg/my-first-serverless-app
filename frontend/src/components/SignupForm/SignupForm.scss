.signupForm {
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding: 10px;
  margin: 0 auto;

  h3{
    text-align: center;
  }

  &__formContainer {
    display: flex;
    flex-direction: column;
  }

  &__email,
  &__password {
    margin: 0 0 10px 0;
    border: 1px solid #cecece;
    padding: 5px;

  }

  &__buttons {
    display: flex;
    flex-direction: row;


  }

  &__loginButton,
  &__signupButton {
    flex: 1;
    padding: 5px;
  }
}