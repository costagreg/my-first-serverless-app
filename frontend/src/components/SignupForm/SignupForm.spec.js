import React from 'react'
import SignupForm from './SignupForm'
import { render } from '@testing-library/react'

describe('SignupForm', () => {
  describe('@renders', () => {
    it('renders <SignupForm /> without any error', () => {
      const { asFragment } = render(<SignupForm />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
