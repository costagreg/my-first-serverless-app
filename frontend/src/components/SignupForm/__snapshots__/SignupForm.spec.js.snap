// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm @renders renders <SignupForm /> without any error 1`] = `
<DocumentFragment>
  <div
    class="signupForm"
  >
    <form
      class="signupForm__formContainer"
    >
      <input
        class="signupForm__email"
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <input
        class="signupForm__password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <input
        class="signupForm__password"
        placeholder="Confirm Password"
        required=""
        type="password"
        value=""
      />
      <div
        class="signupForm__buttons"
      >
        <button
          class="signupForm__submit"
        >
          Sign up
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
